dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/DimensionDescriptor.java

```diff
@@ -0,0 +1,105 @@
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+import static org.apache.commons.lang3.StringUtils.equalsIgnoreCase;
+import static org.apache.commons.lang3.StringUtils.trimToEmpty;
+import static org.hisp.dhis.common.DimensionalObject.ATTRIBUTEOPTIONCOMBO_DIM_ID;
+import static org.hisp.dhis.common.DimensionalObject.CATEGORYOPTIONCOMBO_DIM_ID;
+import static org.hisp.dhis.common.DimensionalObject.DATA_X_DIM_ID;
+import static org.hisp.dhis.common.DimensionalObject.ORGUNIT_DIM_ID;
+import static org.hisp.dhis.common.DimensionalObject.PERIOD_DIM_ID;
+
+import java.util.List;
+import java.util.Map;
+
+import org.hisp.dhis.common.DimensionType;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class DimensionDescriptor
+{
+    private String value;
+
+    private DimensionType type;
+
+    public boolean hasAbbreviation( final String abbreviation )
+    {
+        return trimToEmpty( abbreviation ).equalsIgnoreCase( trimToEmpty( getValue() ) );
+    }
+
+    public static String retrieveDescriptiveValue( final List<DimensionDescriptor> dimensionDescriptors,
+        final String valueToRetrieve, final Map<String, String> metaData )
+    {
+        String descriptiveValue = EMPTY;
+
+        if ( isNotEmpty( dimensionDescriptors ) )
+        {
+            for ( final DimensionDescriptor dimensionDescriptor : dimensionDescriptors )
+            {
+                if ( dimensionDescriptor.hasAbbreviation( valueToRetrieve ) )
+                {
+                    return !equalsIgnoreCase( dimensionDescriptor.getDescriptiveValue(), EMPTY )
+                        ? dimensionDescriptor.getDescriptiveValue()
+                        : defaultIfEmpty( metaData.get( valueToRetrieve ), valueToRetrieve );
+                }
+            }
+        }
+
+        return descriptiveValue;
+    }
+
+    private String getDescriptiveValue()
```

Comment:
Do you think you could add a description about what is going here in the PR and in Javadoc?

Reply:
Done